pkg/monitortests/node/watchnodes/monitortest.go

```diff
@@ -113,3 +113,26 @@ func unreachableNodeTaint(finalIntervals monitorapi.Intervals) []*junitapi.JUnit
 	}
 	return tests
 }
+
+func didIntervalHappenBetweenMachinePhaseChange(interval monitorapi.Interval, machineIntervals monitorapi.Intervals) bool {
+	isIntervalBetweenAMachinePhase := false
+	// There are a few cases we need to catch.
+	for _, val := range machineIntervals {
+		// case 1:
+		// Interval is between the machine config change - no overlap
+		if interval.From.After(val.From) && interval.To.Before(val.To) {
+			return true
+		}
+		// case 2:
+		// Interval is after machine phase change but it lasts beyond the interval
+		if interval.From.After(val.From) && interval.To.Before(val.To) {
+			return true
+		}
+		// case 3:
+		// Interval is before machine phase change but it ends before the interval ends.
+		if interval.From.Before(val.From) && interval.To.After(val.To) {
+			return true
+		}
```

Comment:
I may be confused, but, these all look to be the same condition?

Reply:
I had to write the logic down on paper but I think its correct.